### PR TITLE
CMake: Inform the user when architecture auto-detection fails

### PR DESCRIPTION
**Pull request by yuriks**

### CODE REVIEW

**Comment by linkmauve** [comment]
Every other modern architecture has sizeof(void*) equal to 8 or more commonly 4, I don’t think any architecture we target will be 16-bit or 128-bit, so that fix looks pointless.

**Comment by yuriks** [comment]
The fix is guarding for when `CMAKE_SIZEOF_VOID_P` is *empty*, which then causes it to silently detect at Win32 and fail during linking later with cryptic messages. This is in response to someone on IRC having an issue related to this.

**Comment by linkmauve** [comment]
Oh, ok.
Maybe add a comment saying that, and in which cases it can be empty.

File: CMakeLists.txt [hunk 1]
```diff
@@ -66,8 +66,11 @@ if (ENABLE_GLFW)
         if (MSVC)
             if (CMAKE_SIZEOF_VOID_P EQUAL 8)
                 set(TMP_ARCH "x64")
-            else()
+            elseif (CMAKE_SIZEOF_VOID_P EQUAL 4)
                 set(TMP_ARCH "Win32")
+            else()
+                set(TMP_ARCH "UNKNOWN")
+                message(SEND_ERROR "Couldn't detect your compiler's architecture, you'll have to manually specific the GLFW library to use. (Try checking CMakeOutput.log to find out why.)")
```
Comment (neobrain) [inline CMakeLists.txt]: *specify (same below)

Comment (neobrain) [inline CMakeLists.txt]: => Punctuation and capitalization: "Couldn't detect your compiler's architecture, you'll have to manually specify the GLFW library to use (try checking CMakeOutput.log to find out why)."

**Comment by neobrain** [comment]
LGTM other than the comments I added.

File: CMakeLists.txt (hunk 1 above)
Comment (yuriks) [inline CMakeLists.txt]: Fixing the typo. That's not how parenthesis really work though:
  - http://www.thepunctuationguide.com/parentheses.html
  - http://blog.apastyle.org/apastyle/2013/03/punctuation-junction-periods-and-parentheses.html

**Comment by yuriks** [comment]
Updated.

File: CMakeLists.txt (hunk 1 above)
Comment (neobrain) [inline CMakeLists.txt]: Eh, fair enough about the parentheses.

**Comment by neobrain** [comment]
Two other points just occurred to me:

Btw, what does SEND_ERROR actually do? Does it make cmake fail or will it just print a highlighted error message? If it's the latter, I think we should fail completely in this case.

Also, wouldn't it be better to instead use some make the user specify TMP_ARCH manually, instead of the GLFW library? Seems like the more logical thing to do, and requires less work in the future if we decide to add other libraries which require architecture-specific treatment.

**Comment by yuriks** [comment]
> SEND_ERROR     = CMake Error, continue processing, but skip generation

If we ever need to do that then `TMP_TOOLSET` can be made a cache variable and the user can change that.

**Comment by yuriks** [comment]
Is this good to go? (Trying to empty the PR queue... :) cc @bunnei

**Comment by neobrain** [comment]
I guess it's good. I'd still prefer to have TMP_TOOLSET be configurable as you suggested, simply because it's the more sensible thing to do and since there's a decent chance this will be required at some point in the future, but ultimately that's somewhat out-of-scope of this PR anyway.

**Comment by bunnei** [comment]
LGTM

**Pull request merged**